crates/ott-common/src/discovery/dns.rs

```diff
@@ -50,3 +63,26 @@ impl ServiceDiscoverer for DnsServiceDiscoverer {
         DiscoveryMode::Polling(Duration::from_secs(10))
     }
 }
+
+mod test {
```

Comment:
Without this, this code will always be compiled. This makes it so that it only compiles when you run `cargo test`.
```suggestion
#[cfg(test)]
mod test {
```